.github/workflows/CI_release.yml

```
@@ -0,0 +1,85 @@
+name: CI on release
+
+on:
+  workflow_dispatch:
+  push:    # Initialize release with 'git push origin tag <tag_name>'
+    tags:  # Actual deployment to testpypi and pypi environments requires review
+      - '20[0-9][0-9].0[1-9].[0-9]+'  # Expected limits: 2000.01.0 - 2099.09.123+
+      - '20[0-9][0-9].1[0-2].[0-9]+'  # Expected limits: 2000.10.0 - 2099.12.123+
+
+jobs:
+  build:
+    name: Build package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: astral-sh/setup-uv@v5
+      - run: uv build .
+      - uses: actions/upload-artifact@v4
+        with:
+          name: dist
+          path: dist
+
+  publish-to-testpypi:
+    needs: build
+    name: Publish to TestPyPI
+    runs-on: ubuntu-latest
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/cotainr
+
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v4  # Checkout because `uv publish` requires pyproject.toml
+    - name: Download all the dists
+      uses: actions/download-artifact@v4
+      with:
+        name: dist
+        path: dist
+    - name: Set up uv
+      uses: astral-sh/setup-uv@v5
+    - name: Publish distribution to TestPyPI
+      run: uv publish --index testpypi
+
+  test-testpypi:
+    name: Test TestPyPI Build
+    needs: publish-to-testpypi
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up uv
+      uses: astral-sh/setup-uv@v5
```

For this one we might want to disable the cache to avoid this warning: https://github.com/joasode/cotainr/actions/runs/13923920111/job/38963444497#step:2:19
```suggestion
      uses: astral-sh/setup-uv@v5
      with:
        enable-cache: false
```